clientv3/kv.go

```diff
@@ -106,6 +118,30 @@ func (kv *kv) Delete(key string) (*DeleteResponse, error) {
 	return (*DeleteResponse)(r.GetResponseDeleteRange()), nil
 }
 
+func (kv *kv) Compact(rev int64) error {
+	for {
+		r := &pb.CompactionRequest{Revision: rev}
+		_, err := kv.remote.Compact(context.TODO(), r)
+		if err == nil {
+			return nil
+		}
+
+		if isRPCError(err) {
```

Comment:
`isRPCError` will probably always precede `switchRemote`, so the `isRPCError` check could be folded into `switchRemote`, but not a big deal